### ISSUE TITLE
Weird behavior on back/close pressed

### ISSUE THREAD

**Issue by hamen** [novoda/simple-chrome-custom-tabs#26]
Hi all and thank you for the useful library.

I'm seeing a weird behavior when I try to go back to the app from the CustomTab. When I tap _back_ or the *X* symbol, the CustomTab goes back for a moment to my app and then it jumps to Chrome. This is not happening if there is no Chrome instance in the recent apps. 

My Activity contains this configuration:

```kotlin
    val fallback = NavigationFallback { url ->
        Toast.makeText(applicationContext, "CustomTabs not supported", Toast.LENGTH_SHORT).show()
        val intent = Intent(Intent.ACTION_VIEW).setData(url).addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
        startActivity(intent)
    }
    val customizer = IntentCustomizer { simpleChromeCustomTabsIntentBuilder ->
        val icon = BitmapFactory.decodeResource(resources, android.R.drawable.ic_menu_share)
        val shareLabel = getString(R.string.share_menu_item)
        val actionIntent = Intent(this.applicationContext, ShareBroadcastReceiver::class.java)
        val pendingIntent = PendingIntent.getBroadcast(applicationContext, 0, actionIntent, 0);

        simpleChromeCustomTabsIntentBuilder
            .withUrlBarHiding()
            .withDefaultShareMenuItem()
            .withToolbarColor(ContextCompat.getColor(this, R.color.primary))
            .withActionButton(icon, shareLabel, pendingIntent, false)
    }
    override fun openUrl(url: Uri) {
        SimpleChromeCustomTabs.getInstance()
            .withFallback(fallback)
            .withIntentCustomizer(customizer)
            .navigateTo(url, this)
    }
```

Am I missing something? Thank you.

**Comment by juankysoriano** [comment]
Hello Ivan,

Thank you for your message. 

I have been trying to replicate the issue on the extended demo which is provided on this repository with no luck. I have also applied the same configuration as you posted here in the demo, again; no luck.

I am pretty sure there is nothing wrong in your configuration and the error is lying somewhere else. 

Are you able to provide more details on how does your activity look? 

Are you overriding pending transitions? 
Is there any flag associated to the activity? (e.g `flag_activity_no_history`). 

When you say ( ` goes back for a moment` ) is `onResume` actually being called?

I am sorry I cannot help with more information.

Could you also try a raw implementation of `ChromeCustomTabs` ? (without the usage of this library). The intention is to determine wether or not we actually have a problem in the library, otherwise probably we should open an issue on `ChromeCustomTabs` repository.

Good luck, let us know if you have more details.

**Comment by hamen** [comment]
Hi Jaun and thank you for your reply.

I replicated the issue with a _native_ implementation, too, so I moved my focus elsewhere. Your hint about the flags made me check the `AndroidManifest` and **BAM!** For some _obscure_ reason the `Activity` was marked as `singleInstance`. I removed it and now everything is working as expected.

Tricky scenario indeed, but I'm glad we sorted it out!

Thank you again for your support!
